pkg/apis/psmdb/v1/psmdb_defaults.go

```diff
@@ -734,6 +734,16 @@ func (rs *ReplsetSpec) SetDefaults(platform version.Platform, cr *PerconaServerM
 		}
 	}
 
+	if rs.Storage != nil && rs.Storage.Engine == StorageEngineInMemory {
+		encryptionEnabled, err := rs.Configuration.IsEncryptionEnabled()
+		if err != nil {
+			return errors.Wrap(err, "failed to parse replset configuration")
+		}
+		if encryptionEnabled != nil && *encryptionEnabled {
+			return errors.New("inMemory storage engine doesn't support encryption")
+		}
+	}
```

Comment:
In the codebase we already have a function `func isEncryptionEnabled(cr *api.PerconaServerMongoDB, replset *api.ReplsetSpec) (bool, error) {
`. I think we can utilize it for the most part of this logic by returning a verified not nil boolean.

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/1912/commits/23d9ce0543250918a883de73d66e335ab71ef2ee